docker/test_env/besu.go

```diff
@@ -0,0 +1,221 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/testcontext"
+)
+
+const (
+	//TODO use Tate's mirror?
+	BESU_IMAGE_TAG = "23.10"
+)
+
+type Besu struct {
+	EnvComponent
+	ExternalHttpUrl      string
+	InternalHttpUrl      string
+	ExternalWsUrl        string
+	InternalWsUrl        string
+	InternalExecutionURL string
+	ExternalExecutionURL string
+	generatedDataHostDir string
+	chainConfg           *EthereumChainConfig
+	consensusLayer       ConsensusLayer
+	l                    zerolog.Logger
+	t                    *testing.T
+	image                string
+}
+
+func NewBesu(networks []string, chainConfg *EthereumChainConfig, generatedDataHostDir string, consensusLayer ConsensusLayer, opts ...EnvComponentOption) *Besu {
+	g := &Besu{
+		EnvComponent: EnvComponent{
+			ContainerName: fmt.Sprintf("%s-%s", "besu", uuid.NewString()[0:8]),
+			Networks:      networks,
+		},
+		chainConfg:           chainConfg,
+		generatedDataHostDir: generatedDataHostDir,
+		consensusLayer:       consensusLayer,
+		l:                    logging.GetTestLogger(nil),
+		image:                fmt.Sprintf("hyperledger/besu:%s", BESU_IMAGE_TAG),
+	}
+	for _, opt := range opts {
+		opt(&g.EnvComponent)
+	}
+	return g
+}
+
+func (g *Besu) WithImage(imageWithTag string) *Besu {
```

Comment:
@Tofel maybe you could move it to `EnvComponent` and rename to `WithContainerImage()`? Same can be done with `WithContainerVersion`

Reply:
yeah, but can we do it in a separate PR? this one is already biig

Reply:
yes

Reply:
Added to https://smartcontract-it.atlassian.net/browse/TT-743